test/DelimitedDataParser.Test/ParserTest.Reader.cs

```diff
@@ -1892,5 +1893,87 @@ public void ParseReader_HasRows_ReturnsTrueForHeaderAndDataRows()
 
             Assert.True(reader.HasRows);
         }
+
+        [Fact]
+        public void Supports_Varying_Column_Counts_In_Enumerator_With_Column_Headers()
+        {
+            var content = @"Field 1,Field 2,Field 3" + Environment.NewLine
+                + @"Data 1,Data 2" + Environment.NewLine
+                + @"Data 5" + Environment.NewLine
+                + @"Data 6,Data 7,Data 8, data 9";
```

Comment:
Could we modify this test case to use the same example as provided in #22? We don't need three data rows to reproduce the reported issue.